### PR TITLE
fix: Fix design issues on custom filters

### LINKED ISSUE #18928: Design improvements for personal task filters
### Description

Related to https://github.com/camunda/product-hub/issues/2086, https://github.com/camunda/camunda/issues/17689

### TODO

1. Update the font size for the "Filters" headline and the headline for the selected filter to be `$heading-01`

Image

2. Update the font-size of filter list items to be `$body-01`

Image

3. Increase the width of the Deletion dialog so that the text in CTAs fits in one line. The size can be updated from `cds--modal-container--xs` to `cds--modal-container--sm`

Image

4. To be consistent with the item above, increase the width of the Save dialog to `cds--modal-container--sm`.

Image

5. When a custom filter has a very long name, the text in the headline for the list of tasks gets cut early, leaving an empty space between the text and the sorting action. Instead, the headline should extend having only `$spacing-03` spacing between the text and sorting action.

Image

6. For a custom filter with a very long name that doesn't fit in one line, show a tooltip with a full name when the user hover over it in the list of filters.

Image

### Additional Information

---------------------------------------------------------------------------------------------

### Acceptance Criteria

- [ ] Checked if the specific data needs to be provided for the QA process

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description and testable acceptance criteria

### PR DESCRIPTION
**Pull request by vsgoulart**
## Description
Fixes leftover issues from design review from the custom filters

## Related issues

closes #18928 #18813

### CODE REVIEW

**Comment by vsgoulart** [comment]
@volodymyr-melnykc I couldn't make the link items on the left panel clamp on 2 lines because of some layout issues, so I just showed some ellipsis for now on overflowing texts

File: tasklist/client/src/Tasks/CollapsiblePanel/ControlledNavLink/index.tsx [hunk 1]
```diff
@@ -38,6 +42,7 @@ const ControlledNavLink: React.FC<Props> = ({
           className,
         )
       }
+      title={children.length > 17 ? children : undefined}
```
Comment (Skaiir) [inline tasklist/client/src/Tasks/CollapsiblePanel/ControlledNavLink/index.tsx:45]: What's this doing exactly?

Comment (vsgoulart) [reply tasklist/client/src/Tasks/CollapsiblePanel/ControlledNavLink/index.tsx:45]: @Skaiir  It's showing a native browser tooltip when the text cuts with an ellipsis

**Review by volodymyr-melnykc — APPROVED** [review]
LGTM

File: tasklist/client/src/Tasks/CollapsiblePanel/ControlledNavLink/index.tsx [hunk 2]
```diff
@@ -38,6 +44,7 @@ const ControlledNavLink: React.FC<Props> = ({
           className,
         )
       }
+      title={children.length > ELLIPSIS_CUTOFF_LENGTH ? children : undefined}
```
Comment (vsgoulart) [inline tasklist/client/src/Tasks/CollapsiblePanel/ControlledNavLink/index.tsx:47]: @Skaiir I used a constant to make it more clear

**Review by douglasbouttell-camunda — APPROVED** [review]

**Pull request merged**